shell/browser/api/electron_api_web_contents.cc

```diff
@@ -1086,6 +1095,57 @@ void WebContents::Invoke(bool internal,
                  std::move(callback), internal, channel, std::move(arguments));
 }
 
+void WebContents::ReceivePostMessage(const std::string& channel,
+                                     blink::TransferableMessage message) {
+  auto wrapped_ports = MessagePort::EntanglePorts(isolate(), message.ports);
+  v8::Local<v8::Value> message_value =
+      electron::DeserializeV8Value(isolate(), message);
+  EmitWithSender("-ipc-ports", bindings_.dispatch_context(), InvokeCallback(),
+                 false, channel, message_value, std::move(wrapped_ports));
+}
+
+void WebContents::PostMessage(gin::Arguments* args) {
+  std::string channel;
+  if (!args->GetNext(&channel)) {
+    args->ThrowError();
+    return;
+  }
+
+  v8::Local<v8::Value> message_value;
+  if (!args->GetNext(&message_value)) {
+    args->ThrowError();
+    return;
+  }
+
+  v8::Local<v8::Value> transferables;
+  std::vector<gin::Handle<MessagePort>> wrapped_ports;
+  if (args->GetNext(&transferables)) {
+    if (!gin::ConvertFromV8(isolate(), transferables, &wrapped_ports)) {
+      args->ThrowError();
+      return;
+    }
+  }
+
+  blink::TransferableMessage transferable_message;
+  if (!electron::SerializeV8Value(args->isolate(), message_value,
+                                  &transferable_message)) {
+    args->ThrowError();
```

Comment:
The error would be complaining the 3rd arg instead of 2rd arg because of the `GetNext` above, can probably use `ThrowTypeError` to throw a custom error.

Reply:
Switched the ordering instead, and added a test.